python/ray/train/constants.py

```diff
@@ -49,3 +49,12 @@
 # Backend.share_cuda_visible_devices. 1 for True, 0 for False.
 ENABLE_SHARE_CUDA_VISIBLE_DEVICES_ENV =\
     "TRAIN_ENABLE_SHARE_CUDA_VISIBLE_DEVICES"
+
+# Integer value which indicates the number of seconds to wait when creating
+# the worker placement group before timing out.
+TRAIN_PLACEMENT_GROUP_TIMEOUT_S_ENV = "TRAIN_PLACEMENT_GROUP_TIMEOUT_S"
+
+# Integer value which if set will change the placement group strategy from
+# PACK to SPREAD. 1 for True, 0 for False.
+TRAIN_ENABLE_WORKER_SPREAD_ENV =\
```

Comment:
Looks like I have to set this on all the machines (including HEAD and non-HEAD). Why is that?  Isn't the scheduling part handled by head node (and partially client side)? @matthewdeng

Reply:
Hey @HuangLED, what version of Ray are you using? Also are you using Ray Client?

This should be addressed in master (via [this commit](https://github.com/ray-project/ray/commit/e1cd7b001663f13658ec53425b95697e370f77fb)) so that the environment variable should only need to be set on the driver, but even without this change the environment variable should only be needed on the host where the `BackendExecutor` is.

Reply:
We are using v1.9.0 I think. 

Yes, I am using client mode.  e.g. Two machines M1 and M2, while M1 being the head. Then connecting from a third machine using ray.init("ray://M1:port"). 

Solely setting the env on M1 does not work.

Reply:
Can you verify that the `BackendExecutor` process is running on M1?